src/impression.js

```diff
@@ -102,14 +104,24 @@ function trackImpressionFromClickParam(win, viewer, resolveImpression) {
   }
 
   viewer.whenFirstVisible().then(() => {
+    viewer.sendMessageAwaitResponse('getReplaceUrl', undefined).then(
+        response => {
+          dev().assert(response && typeof response == 'object',
+              'getReplaceUrl expect JsonObject response');
+          if (response['replaceUrl']) {
+            replaceUrl(win, response['replaceUrl']);
+          }
+          resolveImpression();
+        }, () => {});
+
     // TODO(@zhouyx) need test with a real response.
-    const promise = invoke(win, dev().assertString(clickUrl)).then(response => {
-      applyResponse(win, viewer, response);
+    invoke(win, dev().assertString(clickUrl)).then(response => {
```

Comment:
I'm a bit puzzled. Do we want to use this together with the `getReplaceUrl` scheme? Aren't these alternatives?

Reply:
`clickUrl` and `replaceUrl` are used together.